### PR TITLE
Add setup and build instructions for Windows platform

### PR DESCRIPTION
**Pull request by fassisrosa**

### CODE REVIEW

**Comment by fassisrosa** [comment]
This allows me to build LOM on Windows. There are still issues compiling DDR so I'll look at those next and will add to this PR when ready.

File: main/midnight/src/ui/uielement.h
Comment (fassisrosa) [inline main/midnight/src/ui/uielement.h:9]: Ugly but was force to allow the typedef definitions outside the 'ifndef' nlock. VS would not find them otherwise. Wrapped around _OS_WINDOWS_BUILD_ to not break other platform builds.

**Comment by IcemarkUK** [comment]
> This allows me to build LOM on Windows. There are still issues compiling DDR so I'll look at those next and will add to this PR when ready.

Might be worth just getting this PR merged first and worry about DDR after. I don't remember when I last built DDR myself so I might still need to check it works on the Mac! :)

Anyway, great work. Less changes than I expected. I'll pull the request tomorrow and check it out.

File: main/midnight/src/ui/uielement.h
Comment (IcemarkUK) [reply main/midnight/src/ui/uielement.h:9]: Is the position fixed here? This works pretty much anywhere on Mac. So it doesn't need to be wrapped in _OS_WINDOWS_BUILD_ 

This would be ideal if it builds on Windows unless it's the  mxtypes.h which breaks it, in which case it could go between the two. 

```
    #ifndef __UIELEMENT_H_INCLUDED__
    #define __UIELEMENT_H_INCLUDED__

    #include "../cocos.h"
    #include "../library/inc/mxtypes.h"

    // Windows build needs definition upfront
    namespace chilli {
        namespace ui {

            typedef cocos2d::ui::AbstractCheckButton::ccWidgetClickCallback WidgetClickCallback;
            typedef cocos2d::ui::AbstractCheckButton::ccWidgetEventCallback WidgetEventCallback;
        }
    }

    FORWARD_REFERENCE(uieventargs);
    FORWARD_REFERENCE(uielement);
    FORWARD_REFERENCE(uinotificationinterface);
```

**Review by IcemarkUK — COMMENTED** [review]
is proj.win32/game.aps required in the repos? Not sure what it is with it being a binary file.

File: main/midnight/src/ui/uielement.h
Comment (fassisrosa) [reply main/midnight/src/ui/uielement.h:9]: It's not the order that matters. What seems to be happening is that the '#ifndef' declaration at top of .h file is making the typedef definition not be visible in uihelper.h 'WidgetEventCallback' use.  If this was a class definition I could use forward reference for class in 'uihelper.h' but since this is a typedef I cannot. Since this is a typedef declaration it *should* be fine to define it outside the #ifndef block for all platforms (if you're ok with it).

Bottom line: typedef is not visible in uihelper.h when inside the #ifndef (circularity present somehwere?).

File: main/midnight/proj.win32/main.cpp
Comment (fassisrosa) [inline main/midnight/proj.win32/main.cpp:26]: Changed required for VS to find file.

File: main/midnight/src/system/moonring.cpp
Comment (fassisrosa) [inline main/midnight/src/system/moonring.cpp:579]: Would you be ok if his cheat was added to 'default' cheat list? Really handy... ;)

File: main/midnight/src/ui/uielement.h
Comment (fassisrosa) [inline main/midnight/src/ui/uielement.h:214]: Reworked this. Moved typedef definition to bottom of file, not protected by #ifndef ... Issue is related to not being able to forward declare typedef the same way we can forward declare classes.

uihelper.h will not compile without typedef being put outside #ifndef block in this header file.

**Comment by fassisrosa** [comment]
I will not merge this to master without your code review approval. :)

File: main/midnight/src/system/resolutionmanager.h
Comment (fassisrosa) [inline main/midnight/src/system/resolutionmanager.h:79]: I added 'DIS' macro to have it compile. Not clear what DIS should be. Can you explain what this should be?

File: main/midnight/src/tme/scenarios/ddr/ddr_character.cpp
Comment (fassisrosa) [inline main/midnight/src/tme/scenarios/ddr/ddr_character.cpp:21]: include required for std::unique_ptr

File: main/midnight/src/tme/scenarios/ddr/ddr_character.cpp
Comment (fassisrosa) [inline main/midnight/src/tme/scenarios/ddr/ddr_character.cpp:887]: I thought it would be ok to use std::max like other parts of code instead of MAX macro that was not defined. Same for MIN later down the code.

File: main/midnight/src/system/moonring.cpp
Comment (IcemarkUK) [reply main/midnight/src/system/moonring.cpp:579]: Yes, this is fine. Also check out configmanager.

you can put a file in the same folder as settings are stored Can't remember where that is on windows. 

```
<?xml version="1.0" encoding="ISO-8859-1" ?>
<main>
  <variables>
    <var id="cheat_always_win_fight" value="false" />

    <var id="skip_dedication" value="true" /> 
  </variables>
</main>
```

File: main/midnight/src/ui/uielement.h
Comment (IcemarkUK) [reply main/midnight/src/ui/uielement.h:214]: That's ok.

File: main/midnight/src/system/resolutionmanager.h
Comment (IcemarkUK) [reply main/midnight/src/system/resolutionmanager.h:79]: It should just be RES. I must have missed some places. At some stage I had a difference between RES and DIS - possibly font related. But not needed anymore.

**Review by IcemarkUK — APPROVED** [review]

**Pull request merged**
